x/interchainstaking/keeper/ibc_handlers.go

```diff
@@ -180,33 +181,84 @@ func (k *Keeper) HandleCompleteMultiSend(ctx sdk.Context, msg sdk.Msg) error {
 	return nil
 }
 
+// TODO: rework to reflect changes to HandleWithdrawRewards:
+//   1. handle MsgSend from WithdrawalAccount to FeeAccount;
+//   2. handle MsgSend from WithdrawalAccount to Delegation Accounts;
 func (k *Keeper) HandleCompleteSend(ctx sdk.Context, msg sdk.Msg) error {
 	k.Logger(ctx).Info("Received MsgSend acknowledgement")
 	// first, type assertion. we should have banktypes.MsgSend
-	var err error = nil
-	var done bool = false
 	sMsg, ok := msg.(*banktypes.MsgSend)
 	if !ok {
-		k.Logger(ctx).Error("unable to cast source message to MsgSend")
-		return fmt.Errorf("unable to cast source message to MsgSend")
+		err := fmt.Errorf("unable to cast source message to MsgSend")
+		k.Logger(ctx).Error(err.Error())
+		return err
+	}
+
+	// get zone
+	zone, ok := k.GetRegisteredZoneInfo(ctx, ctx.ChainID())
```

Comment:
Just found this - posting here for posterity - `ctx.ChainID()` will always give the local chain ID. We need to do `k.GetZoneForDelegateAccount()` or similar here.